cmd/tools/vdoc/resources/doc.js

```diff
@@ -80,29 +80,89 @@ function setupDarkMode() {
+                var title = searchModuleIndex[i];
+                if (title.indexOf(searchValue) === -1) {
+                    continue
                 }
-                if (li > 0) {
-                    if (!searchValue || link.text.toLowerCase().indexOf(searchValue) !== -1) {
-                        link.style.display = '';
-                    } else {
-                        link.style.display = 'none';
-                    }
+                foundModule = true;
+                // [description, link]
+                var data = searchModuleData[i];
+                var description = data[0];
+                var link = data[1];
+                search.innerHTML += '<ul>' +
+                    '<li class="result">' +
+                        '<a class="link" href="' + link + '">' +
+                            '<div class="definition">' +
+                                '<span class="title">' + title + '</span>' +
+                                '<span class="badge">module</span>' +
+                            '</div>' +
+                            (description ? '<div class="description">' + description + '</div>' : '') +
+                        '</a>' +
+                    '</li>' +
+                '</ul>';
+            }
+            if (foundModule) {
+                search.innerHTML += '<hr class="separator">';
+            }
+            var searchIndexLength = searchIndex.length;
+            var results = [];
+            for (var i = 0; i < searchIndexLength; i++) {
+                var title = searchIndex[i].toLowerCase();
+                if (title.indexOf(searchValue) === -1) {
+                    continue
+                }
+                // [badge, description, link]
+                var data = searchData[i];
+                var badge = data[0];
+                var description = data[1];
+                var link = data[2];
+                var fullName = data[3];
+                results.push({badge, description, link, fullName});
+            }
+            results.sort(function(a, b) {
+                if (a.fullName < b.fullName) {
+                    return -1;
                 }
+                if (a.fullName > b.fullName) {
+                    return 1;
+                }
+                return 0;
+            });
+            for (var i = 0; i < results.length; i++) {
+                var result = results[i];
+                search.innerHTML += '<ul>' +
+                    '<li class="result">' +
+                        '<a class="link" href="' + result.link + '">' +
+                            '<div class="definition">' +
+                                '<span class="title">' + result.fullName + '</span>' +
+                                '<span class="badge">' + result.badge + '</span>' +
+                            '</div>' +
+                            (result.description ? '<div class="description">' + result.description + '</div>' : '') +
```

Comment:
this entire block is pretty much begging for an XSS injection :D

Not a huge issue in this case (since the contents are known beforehand), but I feel like some people may criticise this, and write blog posts about "V's terrible security mechanisms", lol

Plus, it isn't much more complex to write more "correct" code here.

Reply:
Yeah, then tell me how to do frontend rendering instead. This is the only way to do it. Who wants to xss it's own site?
To make an xss you have to get access to the console.
Do you want to include xss detection for such a simple task?

Reply:
`document.createElement()` / `element.append()` ?

Reply:
I can do it. Is it more safe? Just wondering...

Reply:
Yes, it'd be a lot safer. just call e.g. `const node = document.createTextNode(result.whatever)`, and append that node as a child, and that'll avoid any possible xss.

Reply:
yep, or just innerText instead of innerHtml

Reply:
yep

Reply:
fixed it

Reply:
thanks :)